src/tools/tests/schema2kotlin-test.ts

```diff
@@ -256,4 +256,53 @@ describe('schema2kotlin', () => {
     const components = schema2kotlin.generateParticleClassComponents(particle, generators);
     assert.deepEqual(extractor(components), expectedValue);
   }
+
+  describe('Test Harness', () => {
+    it('exposes a handle as a read write handle regardless of particle spec direction', async () => await assertTestHarness(
+      `particle P
+        h1: reads Person {name: Text}
+      `, `
+class PTestHarness<P : AbstractP>(
+    factory : (CoroutineScope) -> P
+) : BaseTestHarness<P>(factory, listOf(
+    HandleSpec("h1", HandleMode.ReadWrite, HandleContainerType.Singleton, P_H1, HandleDataType.Entity)
+)) {
+    val h1: ReadWriteSingletonHandle<P_H1> by handleMap
+}
+`
+    ));
+    it('specifies handle type correctly - singleton, collection, entity, reference', async () => await assertTestHarness(
```

Comment:
What about tuples?

Reply:
(and combinations of other concepts w/ tuples)

Reply:
Haha, added, but they are semi-broken right now, which I will be fixing in the next CL, which is nice, because I will be able to improve the test :)